cmd/experimental/kjobctl/apis/v1alpha1/application_profile_types.go

```diff
@@ -56,19 +70,34 @@ const (
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('error' in self.requiredFlags)", message="error flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('gpus-per-task' in self.requiredFlags)", message="gpus-per-task flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('input' in self.requiredFlags)", message="input flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('job-name' in self.requiredFlags)", message="job-name flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('mem-per-cpu' in self.requiredFlags)", message="mem-per-cpu flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('mem-per-gpu' in self.requiredFlags)", message="mem-per-gpu flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('mem-per-task' in self.requiredFlags)", message="mem-per-task flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('nodes' in self.requiredFlags)", message="nodes flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('ntasks' in self.requiredFlags)", message="ntasks flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('output' in self.requiredFlags)", message="output flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name == 'Slurm' || self.name != 'Slurm' && !('partition' in self.requiredFlags)", message="partition flag can be used only on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name != 'Slurm' || self.name == 'Slurm' && !('parallelism' in self.requiredFlags)", message="parallelism flag can't be used on Slurm mode"
+// +kubebuilder:validation:XValidation:rule="!has(self.requiredFlags) || self.name != 'Slurm' || self.name == 'Slurm' && !('completions' in self.requiredFlags)", message="completions flag can't be used on Slurm mode"
 type SupportedMode struct {
```

Comment:
Add XValidation for mutual exclusive flags (mem-per-task, mem-per-cpu, mem-per-gpu).

Reply:
I got `estimated rule cost exceeds budget` error when adding validation. I tried with two implementations and both failed :

1. `!has(self.requiredFlags) || size(self.requiredFlags.filter(f, f in ['mem-per-task', 'mem-per-cpu', 'mem-per-gpu'])) <= 1`
2. `!has(self.requiredFlags) || !('mem-per-task' in self.requiredFlags || 'mem-per-cpu' in self.requiredFlags || 'mem-per-gpu' in self.requiredFlags) || ('mem-per-task' in self.requiredFlags && !('mem-per-cpu' in self.requiredFlags || 'mem-per-gpu' in self.requiredFlags)) || ('mem-per-cpu' in self.requiredFlags && !('mem-per-task' in self.requiredFlags || 'mem-per-gpu' in self.requiredFlags)) || ('mem-per-gpu' in self.requiredFlags && !('mem-per-task' in self.requiredFlags || 'mem-per-cpu' in self.requiredFlags))`

Reply:
Ah, I think it's because we increased MaxItems...